File: backend/app/schemas/auto_summary.py
from datetime import datetime, date
from typing import Optional, List, Literal
from uuid import UUID
from pydantic import BaseModel


PeriodType = Literal["weekly", "monthly"]
MoodTrend = Literal["improving", "stable", "declining", "mixed"]


class AutoSummaryCreate(BaseModel):
    period_type: PeriodType
    period_start: date
    period_end: date
    title: str
    content: str
    mood_trend: Optional[MoodTrend] = None
    key_themes: Optional[str] = None
    goal_progress: Optional[str] = None
    entry_count: int = 0


class AutoSummaryResponse(BaseModel):
    id: UUID
    period_type: str
    period_start: date
    period_end: date
    title: str
    content: str
    mood_trend: Optional[str]
    key_themes: Optional[str]
    goal_progress: Optional[str]
    entry_count: int
    created_at: datetime

    class Config:
        from_attributes = True


class AutoSummaryListResponse(BaseModel):
    summaries: List[AutoSummaryResponse]
    total: int


class GenerateSummaryRequest(BaseModel):
    period_type: PeriodType


class GenerateSummaryResponse(BaseModel):
    summary: AutoSummaryResponse
    is_new: bool

File: ios/JournalBuddy/JournalBuddy/Models/AutoSummary.swift
import Foundation

enum PeriodType: String, Codable {
    case weekly
    case monthly
}

enum MoodTrend: String, Codable {
    case improving
    case stable
    case declining
    case mixed
}

struct AutoSummary: Codable, Identifiable {
    let id: UUID
    let periodType: PeriodType
    let periodStart: Date
    let periodEnd: Date
    let title: String
    let content: String
    var moodTrend: MoodTrend?
    var keyThemes: String?
    var goalProgress: String?
    let entryCount: Int
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case periodType = "period_type"
        case periodStart = "period_start"
        case periodEnd = "period_end"
        case title, content
        case moodTrend = "mood_trend"
        case keyThemes = "key_themes"
        case goalProgress = "goal_progress"
        case entryCount = "entry_count"
        case createdAt = "created_at"
    }
}

struct AutoSummaryListResponse: Codable {
    let summaries: [AutoSummary]
    let total: Int
}

struct GenerateSummaryResponse: Codable {
    let summary: AutoSummary
    let isNew: Bool

    enum CodingKeys: String, CodingKey {
        case summary
        case isNew = "is_new"
    }
}
